sql-plugin/src/main/scala/org/apache/spark/sql/rapids/GpuFileFormatDataWriter.scala

```diff
@@ -500,7 +506,9 @@ class GpuDynamicPartitionDataSingleWriter(
     }
     val splits = closeOnExcept(cbKeys) { _ =>
       withResource(outputColumnsTbl) { _ =>
-        outputColumnsTbl.contiguousSplit(partitionIndexes: _*)
+        withRetryNoSplit {
```

Comment:
isn't outputColumnsTbl a Table?  Then would that not mean that we would be blocked/retry with memory that is not spillable on the GPU?

Reply:
sorry yeah that `withRetryNoSplit` makes no sense here, it needs to be higher level.. let me fix.

Reply:
I think we actually want to make the table spillable before calling `contiguousSplit` in a `withRetryNoSplit` block and then make those splits spillable.. etc.